### ISSUE TITLE
Keras Multi-Class Accuracy/Loss during testing

### ISSUE THREAD

**Issue by Riashat** [fchollet/keras#2644]
Hi,

How do I measure the accuracy of my multi-class classifier with Keras at testing?

I have a set of predicted labels, and true labels for my image data (multi-class).

Is it by using the model.evalaute(..) in keras? I am using the Sequential model.

I don't quite understand what metrics does the model.evaluate() return.

Thanks

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by sschnug** [comment]
The [docs](http://keras.io/models/sequential/) give some examples:

```
model.compile(optimizer='rmsprop',
          loss='categorical_crossentropy',
          metrics=['accuracy'])  # metric which will be used is defined here
```

And regarding evaluates return values:
```
Scalar test loss (if the model has no metrics) or list of scalars (if the model computes other metrics). The attribute model.metrics_names will give you the display labels for the scalar outputs.
```

**Comment by OverLordGoldDragon** [comment]
For raw per-class _validation_ accuracy:


```
def per_class_accuracy(y_preds,y_true,class_labels):
    return [np.mean([
        (y_true[pred_idx] == np.round(y_pred)) for pred_idx, y_pred in enumerate(y_preds) 
      if y_true[pred_idx] == int(class_label)
                    ]) for class_label in class_labels]

def update_val_history():
    [val_history[class_label].append(np.mean( np.asarray(temp_history).T[class_idx] )
                             ) for class_idx, class_label in enumerate(class_labels)]
```
Example:
```
class_labels = ['0','1','2','3']
val_history = {class_label:[] for class_label in class_labels}

y_true   = np.asarray([0,0,0,0, 1,1,1,1, 2,2,2,2, 3,3,3,3])
y_preds1 = np.asarray([0,3,3,3, 1,1,0,0, 2,2,2,0, 3,3,3,3])
y_preds2 = np.asarray([0,0,3,3, 0,1,0,0, 2,2,2,2, 0,0,0,0])

y_preds1 = model.predict(x1)
temp_hist.append(per_class_accuracy(y_preds1,y_true,class_labels))
update_val_history()
y_preds2 = model.predict(x2)
temp_hist.append(per_class_accuracy(y_preds2,y_true,class_labels))
update_val_history()

print(val_history)
```
**>>{<br>
'0': [0.25, 0.50], <br>
'1': [0.50, 0.25], <br>
'2': [0.75, 1.00], <br>
'3': [1.00, 0.00]  <br>
}**
